utils/src/main/java/io/atomix/utils/serializer/KryoIOPool.java

```diff
@@ -0,0 +1,35 @@
+package io.atomix.utils.serializer;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.function.Function;
+
+class KryoIOPool<T> {
```

Comment:
@NathanSweet have you considered pooling Input and Output objects in Kryo?

Reply:
I think it's application level logic to manage Input and Output allocations.

Reply:
To me seemed more logical to keep it with KyroPool but thanks for the feedback.

Reply:
That is true. FWIW, it wasn't me who added KryoPool. ;)